modules/apps/callbacks/fee_transfer_test.go

```diff
@@ -0,0 +1,199 @@
+
+			fee := feetypes.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+
+			s.ExecutePayPacketFeeMsg(fee)
+			preRelaySenderBalance := sdk.NewCoins(s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), ibctesting.TestCoin.Denom))
+			s.ExecuteTransfer(tc.transferMemo)
+			// we manually subtract the transfer amount from the preRelaySenderBalance because ExecuteTransfer
+			// also relays the packet, which will trigger the fee payments.
+			preRelaySenderBalance = preRelaySenderBalance.Sub(ibctesting.TestCoin)
+
+			// after incentivizing the packets
+			s.AssertHasExecutedExpectedCallbackWithFee(tc.expCallback, tc.expSuccess, false, preRelaySenderBalance, fee)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) TestIncentivizedTransferTimeoutCallbacks() {
+	testCases := []struct {
+		name         string
+		transferMemo string
+		expCallback  types.CallbackType
+		expSuccess   bool
+	}{
+		{
+			"success: transfer with no memo",
+			"",
+			"none",
+			true,
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true, // timeouts don't reach destination chain execution
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTypeTimeoutPacket,
+			true,
+		},
+		{
+			"success: dest callback with low gas (panic)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			"none", // timeouts don't reach destination chain execution
+			false,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			types.CallbackTypeSendPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.SetupFeeTransferTest()
+
+			fee := feetypes.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+
+			s.ExecutePayPacketFeeMsg(fee)
+			preRelaySenderBalance := sdk.NewCoins(s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), ibctesting.TestCoin.Denom))
+			s.ExecuteTransferTimeout(tc.transferMemo, 1)
+
+			// after incentivizing the packets
+			s.AssertHasExecutedExpectedCallbackWithFee(tc.expCallback, tc.expSuccess, true, preRelaySenderBalance, fee)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) ExecutePayPacketFeeMsg(fee feetypes.Fee) {
+	msg := feetypes.NewMsgPayPacketFee(
+		fee, s.path.EndpointA.ChannelConfig.PortID, s.path.EndpointA.ChannelID,
+		s.chainA.SenderAccount.GetAddress().String(), nil,
+	)
+
+	// fetch the account balance before fees are escrowed and assert the difference below
+	preEscrowBalance := s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+
+	res, err := s.chainA.SendMsgs(msg)
+	if err != nil {
+		return // we return if send packet is rejected
+	}
```

Comment:
I don't understand this? Wasn't this just a `require.NoError(err)` before? 

Why would we just want to return here? Also the comment is incorrect, `MsgPayPacketFee` doesn't call `SendPacket` - it just goes directly to the `29-fee` msg server and escrows a fee for the next send sequence send.

Reply:
My bad, I'll revert it

Reply:
All g! I think its unlikely we'd ever really get an error back here from `SendMsgs` anyways 😅 
But yeah, the comment on send packet threw me a little :)